src/relations/backend_db_admin.py

```diff
@@ -0,0 +1,198 @@
+│           │            user │                       │ jujuadmin_pgbouncer-operator          │
+│           │         version │                       │ 12                                    │
+└───────────┴─────────────────┴───────────────────────┴───────────────────────────────────────┘
+
+"""
+
+import logging
+from typing import List
+
+from charms.pgbouncer_operator.v0 import pgb
+from charms.pgbouncer_operator.v0.pgb import PgbConfig
+from ops.charm import (
+    CharmBase,
+    RelationBrokenEvent,
+    RelationChangedEvent,
+    RelationDepartedEvent,
+)
+from ops.framework import Object
+from ops.model import Unit
+
+logger = logging.getLogger(__name__)
+
+RELATION_ID = "backend-db-admin"
+STANDBY_PREFIX = "pgb_postgres_standby_"
+
+
+class BackendDbAdminRequires(Object):
+    """Defines functionality for the 'requires' side of the 'backend-db-admin' relation.
+
+    Hook events observed:
+        - relation-changed
+        - relation-departed
+        - relation-broken
+    """
+
+    def __init__(self, charm: CharmBase):
+        super().__init__(charm, RELATION_ID)
+
+        self.framework.observe(charm.on[RELATION_ID].relation_changed, self._on_relation_changed)
+        self.framework.observe(charm.on[RELATION_ID].relation_departed, self._on_relation_departed)
+        self.framework.observe(charm.on[RELATION_ID].relation_broken, self._on_relation_broken)
+
+        self.charm = charm
+
+    def _on_relation_changed(self, change_event: RelationChangedEvent):
+        """Handle backend-db-admin-relation-changed event.
+
+        Takes master and standby information from the postgresql leader unit databag and copies it
+        into the pgbouncer.ini config, removing redundant standby information along the way.
+        """
+        logger.info("database change detected - updating config")
+        logger.warning(
+            "DEPRECATION WARNING - backend-db-admin is a legacy relation, and will be deprecated in a future release. "
+        )
+
+        postgres_data = change_event.relation.data[change_event.unit]
+
+        cfg = self.charm._read_pgb_config()
+        dbs = cfg["databases"]
+
+        if postgres_data.get("master"):
+            dbs["pg_master"] = pgb.parse_kv_string_to_dict(postgres_data.get("master"))
+        else:
+            logger.info("waiting for postgres charm to correctly populate relation data")
+            change_event.defer()
+            return
+
+        standbys_str = postgres_data.get("standbys")
+        standbys = standbys_str.split("\n") if standbys_str else []
+
+        self._update_standbys(cfg, standbys)
+
+        self.charm._render_pgb_config(cfg, reload_pgbouncer=True)
+
+    def _on_relation_departed(self, departed_event: RelationDepartedEvent):
+        """Handle backend-db-admin-relation-departed event.
+
+        Removes unit information from pgbouncer config when a unit is removed.
+        """
```

Comment:
is it possible that the departing postgres unit was the master? if so, would we need to update `pg_master` key before reloading the config?

also, would we need to screen the departed_event to make sure we're noop-ing if the departing unit is a pgbouncer unit?

Reply:
When a postgres master is removed, the "master" field in the db-admin relation data is updated, causing a relation-changed event. The relation-changed event updates master accordingly, and should actually remove the relevant standbys, so this hook might be irrelevant - I'll look into that. 

Currently there's no pgbouncer scaling implemented (the peer relation is just a stub), so I haven't done your suggestion, but I'll add it in once I add scaling.